[docs] Update using-nextjs.mdx (#22488)
expo/expo: docs/pages/guides/using-nextjs.mdx

> import { Collapsible } from '~/ui/components/Collapsible';

Reviewer: ```suggestion
import { Collapsible } from '~/ui/components/Collapsible';
import { Terminal } from '~/ui/components/Snippet';
```

Adding the statement to import `<Terminal>` component since its used in the document below.

> **Warning:** Using Next.js is not an official part of Expo's universal app development workflow.

> [Next.js][nextjs] is a React framework that provides simple page-based routing as well as server-side rendering. To use Next.js with the Expo SDK, we recommend that you use a library called [`@expo/next-adapter`][next-adapter] to handle the configuration.

Reviewer: ```suggestion
[Next.js][nextjs] is a React framework that provides simple page-based routing as well as server-side rendering. To use Next.js with the Expo SDK, we recommend using [`@expo/next-adapter`][next-adapter] library to handle the configuration.
```

Using Expo with Next.js means you can share some of your existing components and APIs across your mobile and web app. Next.js has its own CLI that you'll need to use when developing for the web platform, so **you'll need to start your web projects with the Next.js CLI and not with `npx expo start`.**

> Next.js can only be used with Expo for web, this doesn't provide Server-Side Rendering (SSR) for native apps.

## Setup

To get started, create a new project with [the template](https://github.com/expo/examples/tree/master/with-nextjs):

> ```sh

Reviewer: ```suggestion
```

> npx create-expo-app -e with-nextjs

Reviewer: ```suggestion
<Terminal cmd={["$ npx create-expo-app -e with-nextjs"]} />
```

> ```

Reviewer: ```suggestion
```

- **Native**: `npx expo start` -- start the Expo project
- **Web**: `npx next dev` -- start the Next.js project

## Manual setup

> ### Dependencies

Reviewer: ```suggestion
### Install dependencies
```

Improving the section header to signify what needs to be done at this step.

> Ensure you have `expo`, `next`, `@expo/next-adapter` installed in your project. `yarn add expo next @expo/next-adapter`

Reviewer: ```suggestion
Ensure you have `expo`, `next`, `@expo/next-adapter` installed in your project:

<Terminal cmd={["$ yarn add expo next @expo/next-adapter"]} />
```

Let's use the `<Terminal>` component instead of `sh` language code snippet to highlight the commands.

### Transpilation

Configure Next.js to transform language features:

<Collapsible summary={<>Next.js with swc. (Recommended)</>}>
  

>   When using Next.js with SWC, you can configure the `babel.config.js` to only account for native.

Reviewer: ```suggestion
  When using Next.js with SWC, you can configure the **babel.config.js** to only account for native.
```

```js babel.config.js
module.exports = function (api) {
  api.cache(true);
  return {
    presets: ['babel-preset-expo'],
  };
};
```

> You will also have to [force Next.js to use SWC](https://nextjs.org/docs/messages/swc-disabled) by adding the following to your `next.config.js`:

Reviewer: ```suggestion
You will also have to [force Next.js to use SWC](https://nextjs.org/docs/messages/swc-disabled) by adding the following to your **next.config.js**:
```

```js next.config.js
module.exports = {
  experimental: {
    forceSwcTransforms: true,
  },
};
```

</Collapsible>

<Collapsible summary={<>Next.js with Babel. (Not recommended)</>}>
  

>   Adjust your `babel.config.js` to conditionally add `next/babel` when bundling with Webpack for web.

Reviewer: ```suggestion
  Adjust your **babel.config.js** to conditionally add `next/babel` when bundling with Webpack for web.
```

```js babel.config.js
module.exports = function (api) {
  // Detect web usage (this may change in the future if Next.js changes the loader)
  const isWeb = api.caller(
    caller =>
      caller && (caller.name === 'babel-loader' || caller.name === 'next-babel-turbo-loader')
  );
  return {
    presets: [
      // Only use next in the browser, it'll break your native project
      isWeb && require('next/babel'),
      'babel-preset-expo',
    ].filter(Boolean),
  };
};
```

</Collapsible>

### Next.js configuration

> Add the following to your `next.config.js`:

Reviewer: ```suggestion
Add the following to your **next.config.js**:
```

```js next.config.js
const { withExpo } = require('@expo/next-adapter');

module.exports = withExpo({
  // transpilePackages is a Next.js +13.1 feature.
  // older versions can use next-transpile-modules
  transpilePackages: [
    'react-native',
    'expo',
    // Add more React Native / Expo packages here...
  ],
});
```

The fully qualified Next.js config may look like:

```js next.config.js
const { withExpo } = require('@expo/next-adapter');

/** @type {import('next').NextConfig} */
const nextConfig = withExpo({
  reactStrictMode: true,
  swcMinify: true,
  transpilePackages: [
    'react-native',
    'expo',
    // Add more React Native / Expo packages here...
  ],
  experimental: {
    forceSwcTransforms: true,
  },
});

module.exports = nextConfig;
```

### React Native Web styling

> The package `react-native-web` builds on the assumption of reset CSS styles, here's how you reset styles in Next.js using the `pages/` directory.

Reviewer: ```suggestion
The package `react-native-web` builds on the assumption of reset CSS styles. Here's how you reset styles in Next.js using the **pages** directory.
```

  overscroll-behavior-y: none;
  text-rendering: optimizeLegibility;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  -ms-overflow-style: scrollbar;
}
`;

export default class MyDocument extends Document {
  static async getInitialProps({ renderPage }) {
    AppRegistry.registerComponent('main', () => Main);
    const { getStyleElement } = AppRegistry.getApplication('main');
    const page = await renderPage();
    const styles = [
      <style key="react-native-style" dangerouslySetInnerHTML={{ __html: style }} />,
      getStyleElement(),
    ];
    return { ...page, styles: Children.toArray(styles) };
  }

  render() {
    return (
      <Html style={{ height: '100%' }}>
        <Head />
        <body style={{ height: '100%', overflow: 'hidden' }}>
          <Main />
          <NextScript />
        </body>
      </Html>
    );
  }
}
```

```js pages/_app.js
import Head from 'next/head';

export default function App({ Component, pageProps }) {
  return (
    <>
      <Head>
        <meta name="viewport" content="width=device-width, initial-scale=1" />
      </Head>
      <Component {...pageProps} />
    </>
  );
}
```

## Transpiling modules

> By default, modules in the React Native ecosystem are not transpiled to run in web browsers. React Native relies on advanced caching in Metro to reload quickly. Next.js uses Webpack, which does not have the same level of caching, so by default, no node modules are transpiled. You will have to manually mark every module you want to transpile with the `transpilePackages` option in `next.config.js`:

Reviewer: ```suggestion
By default, modules in the React Native ecosystem are not transpiled to run in web browsers. React Native relies on advanced caching in Metro to reload quickly. Next.js uses Webpack, which does not have the same level of caching, so no node modules are transpiled by default. You will have to manually mark every module you want to transpile with the `transpilePackages` option in **next.config.js**:
```

[File names to be bold as per our writing styleguide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md#filenames-as-bold-text).

Also, we are repeating the context of "by default" twice in the paragraph above. Kept the first instance and remove the second one.

- Install the Vercel CLI: `npm i -g vercel`
- Deploy to Vercel: `vercel`

## Limitations or differences comparing to the default Expo for Web

> Using Next.js for the web means you will be bundling with the Next.js Webpack config. This will lead to some core differences in how you develop your app vs your website.

Reviewer: ```suggestion
Using Next.js for the web means you will be bundling with the Next.js webpack config. This will lead to some core differences in how you develop your app vs your website.
```

Fixed the lint case police for "Webpack" -> "webpack".

- Expo Next.js adapter does not support the experimental app directory.
- For File-based routing on native, we recommend using [Expo Router](https://github.com/expo/router).

## Contributing

> If you would like to help make Next.js support in Expo better, please feel free to open a PR or submit an issue:

Reviewer: ```suggestion
If you would like to help make Next.js support in Expo better, feel free to open a PR or submit an issue:
```

- [@expo/next-adapter][next-adapter]

## Troubleshooting

### Cannot use import statement outside a module

> Figure out which module has the import statement and add it to the `transpilePackages` option in `next.config.js`:

Reviewer: ```suggestion
Figure out which module has the import statement and add it to the `transpilePackages` option in **next.config.js**:
```
